src/auth_service/auth_adapter/api/main.py

```diff
@@ -189,6 +208,68 @@ async def logout(
     return Response(status_code=204)
 
 
+@app.post(
+    "/users",
+    operation_id="post_user",
+    tags=["users"],
+    summary="Register a user",
+    description="Authenticate the endpoint used to register a new user.",
+    status_code=200,
+)
+async def post_user(
+    session: SessionDependency,
+    x_csrf_token: Annotated[Optional[str], Header()] = None,
+) -> Response:
+    """Register a user."""
+    response = Response(status_code=200)
+    if not session:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED, detail="Not logged in"
+        )
+    check_csrf("POST", x_csrf_token, session)
+    internal_token = internal_token_from_session(session)
+    authorization = f"Bearer {internal_token}"
+    response.headers["Authorization"] = authorization
+    return response
+
+
+@app.put(
+    "/users/{id}",
+    operation_id="post_user",
+    tags=["users"],
+    summary="Update a user",
+    description="Authenticate the endpoint used to update an existing user.",
```

Comment:
```suggestion
    description="Authenticate against the endpoint used to update an existing user.",
```

Reply:
Actually, I meant to authenticate the endpoint itself (i.e. allowing that the endpoint can be used). Will try to improve the docstring in the next PR.